server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1478,6 +1469,20 @@ ClusterState getStateForMasterService() {
         }
     }
 
+    private ClusterState getStateForJoinValidationService() {
+        synchronized (mutex) {
+            // similar to getStateForMasterService, but do not rebuild the state if not currently the master
+            final ClusterState clusterState = coordinationState.get().getLastAcceptedState();
+            assert clusterState.nodes().getLocalNode() != null;
+            if (mode != Mode.LEADER
+                || clusterState.term() != getCurrentTerm()
+                || clusterState.nodes().isLocalNodeElectedMaster() == false) {
```

Comment:
> `|| clusterState.nodes().isLocalNodeElectedMaster() == false) {`

This line is new and previously it was an assertion. Why is that?

Reply:
I don't see where this used to be an assertion, sorry if I'm missing something. This is really just folding in the `isLocalNodeElectedMaster` check that previously was here:

https://github.com/elastic/elasticsearch/blob/0b5f19bdfcd639134d36674277e748d72d4926d2/server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java#L695